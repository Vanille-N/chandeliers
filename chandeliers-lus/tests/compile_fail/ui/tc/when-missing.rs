//! Undeclared variable used as clock.
chandeliers_lus::decl! {
    node foo(x : int when b) returns ();
    let tel;
}

chandeliers_lus::decl! {
    node foo(x : int when b) returns (b : bool);
    let b = true; tel;
}

fn main() {}
