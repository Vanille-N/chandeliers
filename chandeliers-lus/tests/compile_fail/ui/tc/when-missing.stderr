error: Variable b is not available yet at this point of the typechecking.
 --> tests/compile_fail/ui/tc/when-missing.rs:3:27
  |
3 |     node foo(x : int when b) returns ();
  |                           ^
  |
  = note: During this incremental typechecking, you cannot access global
          variables and you may only use local variables that have been
          declared strictly beforehand, in the order of inputs then outputs
          then locals.
  = note: These are the variables that are already useable: (none declared)

error: Variable b is not available yet at this point of the typechecking.
 --> tests/compile_fail/ui/tc/when-missing.rs:8:27
  |
8 |     node foo(x : int when b) returns (b : bool);
  |                           ^
  |
  = note: During this incremental typechecking, you cannot access global
          variables and you may only use local variables that have been
          declared strictly beforehand, in the order of inputs then outputs
          then locals.
  = note: These are the variables that are already useable: (none declared)
